blosc2/blosc2_ext.pyx

```diff
@@ -655,28 +673,148 @@ cdef class SChunk:
         self.schunk.chunksize = chunksize
         cdef const uint8_t[:] typed_view
         cdef int64_t index
+        cdef Py_buffer *buf
         if data is not None:
-            mem_view = memoryview(data)
-            typed_view = mem_view.cast('B')
-            len_data = typed_view.nbytes
+            buf = <Py_buffer *> malloc(sizeof(Py_buffer))
+            PyObject_GetBuffer(data, buf, PyBUF_SIMPLE)
+            len_data = buf.len
             nchunks = len_data // chunksize + 1 if len_data % chunksize != 0 else len_data // chunksize
             len_chunk = chunksize
             for i in range(nchunks):
                 if i == (nchunks - 1):
                     len_chunk = len_data - i * chunksize
                 index = i * chunksize
-                nchunks_ = blosc2_schunk_append_buffer(self.schunk, <void*>&typed_view[index], len_chunk)
+                nchunks_ = blosc2_schunk_append_buffer(self.schunk, <void*>&buf.buf[index], len_chunk)
                 if nchunks_ != (i + 1):
+                    PyBuffer_Release(buf)
                     raise RuntimeError("An error occurred while appending the chunks")
+            PyBuffer_Release(buf)
 
     @property
     def c_schunk(self):
         return <uintptr_t> self.schunk
 
+    @property
+    def chunkshape(self):
+        """
+        Number of elements per chunk.
+        """
+        return self.schunk.chunksize // self.schunk.typesize
+
+    def __len__(self):
+        return self.schunk.nbytes // self.schunk.typesize
+
     @property
     def chunksize(self):
+        """
+        Number of bytes in  each chunk.
+        """
         return self.schunk.chunksize
 
+    @property
+    def cratio(self):
+        """
+        Compression ratio.
+        """
+        if self.schunk.cbytes == 0:
+            raise ZeroDivisionError("Empty schunk")
+        return self.schunk.nbytes / self.schunk.cbytes
+
+    @property
+    def nbytes(self):
+        """
+        Number of data bytes in the `SChunk`.
+        """
+        return self.schunk.nbytes
+
+    @property
+    def cbytes(self):
+        """
+        Number of compressed bytes.
```

Comment:
"Amount of compressed data bytes"

Reply:
Fixed in 8fe520bc3b8fdaa36991a8037bec4163a263596e.